scarlet/source.py

```diff
@@ -144,10 +144,10 @@ def trim_morphology(sky_coord, frame, morph, bg_cutoff, thresh):
         if bbox.contains(pixel_center):
             size = 2 * max(
                 (
-                    pixel_center[0] - bbox.bottom,
-                    bbox.top - pixel_center[0],
-                    pixel_center[1] - bbox.left,
-                    bbox.right - pixel_center[1],
+                    pixel_center[0] - bbox.start[0],
+                    bbox.stop[0] - pixel_center[0],
+                    pixel_center[1] - bbox.start[1],
+                    bbox.stop[1] - pixel_center[1],
```

Comment:
This is the kind of place that makes me wonder why we want to generalise the dimensionality of the boxes. Here we consider that pixels are in dimensions `(0,1)` (btw, I thought it would be along `(1,2)` or `(-2,-1)`) and it makes me think that we have to think of what those other axes might be and we take them into account.

Reply:
I think that since we insist on the convention `(channels, Ny, Nx)` for 3D images while a 2D image will have dimension `(Ny, Nx)` there is no ambiguity here. Since morphology is always 2D, it's consistent with this convention to use `(0,1)`. But I do see how this can be confusing and wouldn't be opposed to `(-2, -1)` for clarity.

Reply:
It's fully consistent. This method internally creates the box from a 2D image, hence it's 2D. Nobody else touches the bbox, so as long as the author of that piece of code knows how BBox behaves, it's all good.

Reply:
changed to (-2, -1)

Reply:
Yup, I missed the 2D nature of the morphologies, fine. I guess my concern was more about generalising to higher dimensions but since we saw earlier that this is not the intended purpose, it's ok. Still (-2,-1) is nice.